modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/Diagram/Diagram.tsx

```diff
@@ -56,6 +56,19 @@ function DiagramBuilder({
 		dispatch,
 	] = useObjectFolderContext();
 
+	useEffect(() => {
+		if (showChangesSaved) {
+			setTimeout(() => {
+				dispatch({
+					payload: {updatedShowChangesSaved: false},
+					type: TYPES.SET_SHOW_CHANGES_SAVED,
+				});
+			}, 5000);
+		}
+
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [showChangesSaved]);
+
```

Comment:
Hey @Artartur! Thanks for this PR. I just have some suggestions:

My first idea is to try to achieve the expected results without using `useEffect`, making `SET_SHOW_CHANGES_SAVED` take care of its disappearance, this would mean that we would not need to use `useEffect` or make two function calls outside the use reduce case. Something like that:

 case TYPES.SET_SHOW_CHANGES_SAVED: {
 	const {updatedShowChangesSaved, dispatch} = action.payload;

 	if(updatedShowChangesSaved){
 		setTimeout(() => {
 			dispatch({type: TYPES.SET_SHOW_CHANGES_SAVED, payload: {updatedShowChangesSaved: false}});
 		  }, 5000);
 	}

 	return {
 		...state,
 		showChangesSaved: updatedShowChangesSaved,
 	};
 }
But we don't have any example inside the `objectFoldeReducer.tsx `calling itself.

The seccond one is move your `useEffect` to more external file:

![object-folder drawio](https://github.com/liferay-objects/liferay-portal/assets/61854510/2d5a0ba8-af3f-4e03-a65d-8dd3e2508644)

As you can see, `Diagram.tsx` is one of four files that can change the state of the `showChangesSaved` property.

We have an outermost file called `EditObjectFolder`.

This file is responsible for calling all components that can modify the state of this property, so I believe that the ideal place for this `useEffect` would be in the `EditObjectFolder`.

What do you think, @danvb112 @naraandrade?

Reply:
Hey @Artartur, could you try to implement the first solution without useEffect and check that we don't have errors in the terminal and that this solution works in all cases? Thanks!

Reply:
Hi guys, I have an update on this topic. 
This [bug](https://github.com/reduxjs/redux/issues/368#issuecomment-126496428) was opened from Dan Abramov (the creator of Redux). 

In sumary he said,
> "Why would you want to dispatch inside a reducer? It's grossly misusing the library. It's exactly the same as React doesn't allow you to setState inside render."

So, you could follow the second approach @Artartur.

Reply:
Sure, done!

Reply:
I did some research on this subject and it's actually better not to call `dispatch` within the `reducer`, it would be considered bad practice, it would be as if within a React `setState` we called `setState` again, this could lead to a problem using the value not updated, because the `reducer` did not have time to finish the function until we call it again. In this specific case, which is a small function, I don't think we will have this problem, but to maintain best practices it is better to use the `useEffect` approach. :smile: